hardemu/pio.js

```diff
@@ -41,7 +48,12 @@ function PIO(Zeal) {
         dir_follows: false,
         /* Each listener is a callback that needs to be called when a write or read occurs on a pin.
          * So we have a most 8 listeners */
-        listeners: []
+        listeners: [],
+        /* Each listener is a callback and state that needs to be called when a write occurs on a pin.
+         * The callback is only called if the current state matches the state and the previous state does not.
+         * indicating that the pin has "changed into the expected state"
+         * (ie; we want the pin HIGH, and it was previously LOW) */
+        listeners_change: [],
```

Comment:
Is there any particular reason for introducing this type of callbacks and not using the former `listeners`?

Reply:
these are transition hooks - so the listener doesn’t need to implement its own transition handling - as it’s a common task, I figured implementing it here would allow future code to be simpler